### PR TITLE
Updated process_cut() function to add default values to the patient_c…

### PR DESCRIPTION
**Pull request by reesnj**
Hello @kieranjmartin and @barnett11,
As per issue #134 - I have added default values for the following function arguments:

- patient_cut_v = vector()
- date_cut_m = matrix(nrow=0, ncol=2)
- no_cut_v = vector()

In the reproducible examples provided in #134, these updates means that your first example should now work because no_cut_v will be set to the default value (an empty vector). However, your second example will still fail since it is expected that no_cut_v is a vector but you have set this to "" here. Hopefully this is clear from the documentation of the function but please let know if you think further updates are required.

### CODE REVIEW

**Comment by reesnj** [comment]
Hello @barnett11 - thank you for your comments! 
I have updated the assertions to make the error messages more meaningful. Now, if you put in "", this will tell you that the input must be a vector, and also give you a note that if you do not want any domains to use this type of cut, then you should leave the parameter empty, rather than using "". 
Whereas, if you define a cut for an SDTM that is not in the input list, it will give a different error message indicating that there are inconsistencies between the input SDTMv datasets and the SDTMv datasets listed under each cut approach.
Hopefully you find these messages more informative!

File: R/process_cut.R [hunk 1]
```diff
@@ -50,34 +45,41 @@
...
+    msg = "patient_cut_v must be a vector. \n
+Note: If you do not wish to use a patient cut on any SDTMv domains, then please leave
+patient_cut_v empty, in which a case default value of vector() will be used."
   )
-  assert_that(is.matrix(date_cut_m),
-    msg = "date_cut_m must be a matrix"
+  assert_that(all(is.matrix(date_cut_m), date_cut_m != ""),
+    msg = "date_cut_m must be a matrix \n
+Note: If you do not wish to use a date cut on any SDTMv domains, then please leave
+date_cut_m empty, in which a case default value of matrix(nrow=0, ncol=2) will be used."
```
Comment (barnett11) [inline R/process_cut.R:74]:
```suggestion
date_cut_m empty, in which case a default value of matrix(nrow=0, ncol=2) will be used."
```

**Review by barnett11 — CHANGES_REQUESTED** [review]
Looking good @reesnj - small update suggested

**Comment by kieranjmartin** [comment]
Hi, I think this is fine if this is the solution which works, and making it default seems solid to me. That said, I think conventionally in R functions a blank argument would be expected to be NULL; my example with a missing string was probably not a good way to approach it either. 

I'm fine with it as is, however, as it is documented and asserted, so should be fine

**Comment by reesnj** [comment]
Agreed to update this function at the next release to change the default values of function arguments to be NULL in order to follow a more conventional approach (task added to project board). However, agreed to leave this function as is for the current release.

**Pull request merged**